compiler/integration-tests/test/fuzzy/fuzzy.test.ts

```diff
@@ -0,0 +1,106 @@
+  for (let i = 0; i < iterations; i++) {
+    it("should error compile() gracefully", async () => {
+      await newCompiler();
+      await newABICoder();
+      await initACVM();
+      const fuzzedNoirSource = getRandomString(
+        Math.floor(Math.random() * Math.pow(10, i)),
+      );
+      initializeResolver(() => fuzzedNoirSource);
+      let compileOutput;
+      let errorOccurred = false;
+      try {
+        compileOutput = await compile({});
+        expect(compileOutput).to.be.an("object");
+      } catch (e) {
+        expect(e.toString().includes("RuntimeError: unreachable executed")).to
+          .be.true;
+        errorOccurred = true;
+      }
+      expect(
+        errorOccurred,
+        "Compilation should have failed with the fuzzed input",
+      ).to.be.true;
+    });
+  }
+});
+
+describe("abiEncode fuzz test", () => {
+  for (let i = 0; i < iterations; i++) {
+    it(`should error abiEncode() gracefully`, async () => {
+      let errorTriggered = false;
+      const stringLength = Math.pow(10, i);
+      const randomString = getRandomString(stringLength);
+      const fuzzedTOML = `${randomString} = "${randomString}"`;
+      try {
+        const inputs = TOML.parse(fuzzedTOML);
+        expect(inputs).to.be.an("object");
+        abiEncode(randomString, inputs, null);
+      } catch (e) {
+        console.log({ e });
+        expect(e.toString().includes("Error: invalid type")).to.be.true;
+        errorTriggered = true;
+      }
+
+      expect(errorTriggered).to.be.true;
+    });
+  }
+});
+
+// timeouts and errors in acvm_js
+xdescribe("executeCircuit fuzz test", function () {
+  this.timeout(120e3);
+
+  for (let i = 0; i < iterations; i++) {
+    it(`should error executeCircuit() gracefully`, async () => {
+      let errorTriggered = false;
+      const numberLength = Math.pow(10, i);
+      try {
+        const compressedByteCode = getRandomUint8Array(numberLength);
+        const witnessMap = getRandomMap(numberLength);
+        await executeCircuit(compressedByteCode, witnessMap, () => {
+          throw Error("unexpected oracle");
+        }).catch();
+      } catch (e) {
+        expect(e.toString().includes("RuntimeError: unreachable executed")).to
+          .be.true;
+        errorTriggered = true;
+      }
+
+      expect(errorTriggered).to.be.true;
+    });
+  }
+});
+
+describe("compressWitness fuzz test", () => {
+  for (let i = 0; i < iterations; i++) {
+    it(`should error compressWitness() gracefully`, async () => {
+      let errorTriggered = false;
+      const mapLength = Math.pow(10, i);
+      const randomMap = getRandomMap(mapLength);
+      try {
+        const compressedWitness = compressWitness(randomMap);
+        expect(compressedWitness).to.be.an("object");
+      } catch (e) {
+        expect(e.toString().includes("RuntimeError: unreachable executed")).to
+          .be.true;
+        errorTriggered = true;
+      }
+
+      expect(errorTriggered).to.be.true;
+    });
+  }
+});
```

Comment:
This is equivalent to fuzzing the trait implementation below
https://github.com/noir-lang/noir/blob/4deb07f80ce110187b66a46dd5624af3b8df3dbd/acvm-repo/acir/src/native_types/witness_map.rs#L81-L91

If we wanted to fuzz this then I think it would be more effective to do this inside Rust tbh.

Reply:
That said, I don't think this would be a great use of time as we're essentially just fuzz testing `GzEncoder` whereas we can just rely on their test suite.